Source/RageCommunity.Library/Graphics/Checkpoint.cs

```diff
@@ -0,0 +1,223 @@
+﻿using System;
+using System.Drawing;
+using Rage;
+using Rage.Native;
+using RageCommunity.Library.Extensions;
+
+namespace RageCommunity.Library.Graphics
+{
+    /// <summary>
+    /// This class represents a checkpoint within GTA V.
+    /// </summary>
+    /// <remarks>
+    /// Originally created by PNWParksFan. 
+    /// </remarks>
+    public class Checkpoint : IDeletable, ISpatial
+    {
+        private readonly Boolean setToGround;
+        private Color color;
+        private Single height;
+        private Vector3 nextPosition;
+        private Vector3 position;
+
+        public Checkpoint(Vector3 position,
+                          Color color,
+                          Single radius = 1f,
+                          Single height = 1f,
+                          Int32 checkpointType = 47,
+                          Int32 reserved = 0,
+                          Boolean setZtoGround = true) : this(position, position, color, radius, height, checkpointType, reserved, setZtoGround)
+        {
+        }
+
+        public Checkpoint(Vector3 position,
+                          Vector3 nextPosition,
+                          Color color,
+                          Single radius,
+                          Single height,
+                          Int32 checkpointType = 0,
+                          Int32 reserved = 0,
+                          Boolean setZtoGround = true)
+        {
+            CheckpointType = checkpointType;
+            this.position = position;
+            this.nextPosition = nextPosition;
+            Radius = radius;
+            this.color = color;
+            Reserved = reserved;
+            this.setToGround = setZtoGround;
+            this.height = height;
+
+            CreateCheckpoint();
+        }
+
+        /// <summary>
+        /// The checkpoint types represents how the checkpoints looks in the game. 
+        /// </summary>
+        /// <remarks>
+        /// Checkpoint type 47 for example is a simple cylinder with no arrow or number.
+        /// Visit the Grand Theft Auto V Native DB for a list of all checkpoint types. 
```

Comment:
Can we link to a URL with this information? https://nativedb.dotindustries.dev/natives?search=checkpoint

Reply:
Yes, I will follow your advice.